### ISSUE TITLE
class_cast_exception raised when sorting by unnest-ed object column

### ISSUE THREAD

**Issue by shm-dmitry** [crate/crate#10122]
**CrateDB version**: 4.1.6

**Steps to reproduce**:
`create table test (a string, b string, c array(object as (d string, e string)));`
`>> Table TEST created.`

`insert into test (a, b, c) values ('x', 'y', [ { d = '1', e = '2' }, { d = '3', e = '4' } ]);`
`>> 1 row inserted.`

`select a, b, unnest(c['d']), unnest(c['e']) from test order by 1,3;`
**`ERROR: NotSerializableExceptionWrapper: class_cast_exception: class io.crate.expression.tablefunctions.ColumnOrientedRowsIterator cannot be cast to class java.lang.String (io.crate.expression.tablefunctions.ColumnOrientedRowsIterator is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')`**

**Comment by mfussenegger** [comment]
Thank you for the bug report.

This is already fixed in master and a workaround would be to wrap the `unnest` parts into a sub-query and apply the `ORDER BY` outside of it:

```
select * from (select a, b, unnest(c['d']), unnest(c['e']) from test) t order by 1, 3
```

We'll have a look at backporting the fix for the upcoming 4.1.7

**Comment by mfussenegger** [comment]
Should be fixed via https://github.com/crate/crate/pull/10127
